Mathlib/Data/List/Basic.lean

```diff
@@ -395,6 +395,11 @@ theorem cons_eq_append_iff {a b c : List α} {x : α} :
 @[deprecated] alias append_right_cancel := append_cancel_right -- deprecated since 2024-01-18
 #align list.append_right_cancel List.append_cancel_right
 
+@[simp] theorem append_left_cancel_nil_left {x y : List α}: x ++ y = x ↔ y = [] := by
+  constructor <;> intro h; apply List.append_cancel_left (as := x); all_goals simp[h]
+@[simp] theorem append_left_cancel_nil_right {x y : List α}: x = x ++ y ↔ y = [] :=
+  by rw[eq_comm, append_left_cancel_nil_left]
```

Comment:
Per the style guide, please put the `by` on the previous line, and also put a space after the tactic name (i.e., do `rw [eq_comm]` instead of `rw[eq_comm]` etc.)

Reply:
About `rw[...]` vs `rw [...]`: I couldn't find this rule in your style guide (although the examples there comply), but just something about spaces after left arrows, and also had to fix this at places in the file not added by this commit. Maybe you should add it?